internal/controller/prefectserver_controller.go

```diff
@@ -627,30 +639,35 @@ func (r *PrefectServerReconciler) postgresDeploymentSpec(server *prefectiov1.Pre
 }
 
 func (r *PrefectServerReconciler) postgresMigrationJob(server *prefectiov1.PrefectServer) *batchv1.Job {
-	return &batchv1.Job{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: server.Namespace,
-			Name:      server.Name + "-migration",
-		},
-		Spec: batchv1.JobSpec{
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: server.ServerLabels(),
-				},
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{
-						{
-							Name:    "prefect-server-migration",
-							Image:   server.Image(),
-							Command: []string{"prefect", "server", "database", "upgrade", "--yes"},
-							Env:     append(append(server.ToEnvVars(), server.Spec.Postgres.ToEnvVars()...), server.Spec.Settings...),
-						},
+	jobSpec := batchv1.JobSpec{
+		TTLSecondsAfterFinished: ptr.To(int32(7 * 24 * 60 * 60)), // 7 days
```

Comment:
since we're creating a new Job on every PrefectServer update, we'll have Jobs like this:

```
kubectl get jobs

prefect-migration-1234
prefect-migration-5678
prefect-migration-2956
...
```

we'll add a 1-week TTL that cleans up the Job after it is complete - Im thinking this gives enough of a window for retrieving past Job runs and no-op-ing if the Job Spec hasn't changed

Reply:
Perfect!